src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.sql.Timestamp;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
+    private static final DateFormat FORMATTER = new SimpleDateFormat("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        try {
+            Timestamp fromTime = dateParse(dateFrom);
+            Timestamp toTime = dateParse(dateTo);
+            for (String name : names) {
+                int sum = 0;
+                for (String info : data) {
+                    String[] dateArray = info.split(" ");
+                    if (checkDate(dateParse(dateArray[InfoType.DATE.ordinal()]), fromTime, toTime)
+                            && info.contains(name)) {
+                        sum += Integer.parseInt(dateArray[InfoType.WORKING_HOUR.ordinal()])
+                                * Integer.parseInt(dateArray[InfoType.INCOME_PER_HOUR.ordinal()]);
+                    }
+                }
+                result.append(System.lineSeparator())
+                        .append(name)
+                        .append(" - ")
+                        .append(sum);
+            }
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
```

Comment:
lets add to try-catch block only date parse logic, not all method logic. And it's bad practice to use e.printStackTrace() , lets better throw some RuntimeException with informative message instead